update-nav-bar

raymondng1 opened the pull request:
When the user logs in, the navigation bar would change having the components of the user's profile with their name. 

Next steps going forward for the log out functionality, do we want to delete the session ID off the User and creating a new guest in our table (basically reversing the actions of our login- functionality).

rng193 committed:
update-nav-bar

ShelOrock commented on app/components/NavGuest.js:
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
watch linting - i think we're doing 2 spaces not tabs

leafoflegend replied:
Merge the Travis PR first, add linting to CI. Shouldn't have to catch this manually.

nries1 replied:
I'm working on Travis CI now. I'll try to get it up b4 next pr.

ShelOrock commented on app/components/NavUser.js:
```diff
@@ -0,0 +1,20 @@
+import { Nav, Button } from 'react-bootstrap';
+import { connect } from 'react-redux';
+
+const NavUser = ({user})=> {
```
is there a possibility the UserNav and GuestNav could be one component? and the nav link displayed would be different based on the userType using a ternary? You def use a ternary in the navigation component to do this, but I personally prefer components closer to the root to do less work and the outer ones to take care of heavy lifting. Just a thought!

raymondng1 replied:
Yup you definitely have a good point to make it into one component, will definitely update with the userType as well

ShelOrock commented on app/components/Navigation.js:
```diff
@@ -2,9 +2,13 @@ import React, { Component } from 'react';
 import { Nav, Navbar, Button } from 'react-bootstrap';
```
watch linting. i think the original component you were working from was inconsistent

ShelOrock commented on app/components/Navigation.js:
```diff
@@ -2,9 +2,13 @@ import React, { Component } from 'react';
 class Navigation extends Component {
 	render() {
+		const { user } = this.props;
+		const links = user.firstName ? <NavUser/> : <NavGuest />;
```
I would check this using a switch on the userType. We may want to render a different nav for admins as well who will have a first name. Also we can store other cookies (yikes) that would auto-populate shipping fields for example and we could store a name so theoretically a Guest could have a firstName as well.

ShelOrock reviewed (commented):
The code makes logical sense to me. Nice job. Is there a way we can test this? For front end components we should def include acceptance criteria so we can all replicate your intent.

rng193 committed:
updated with switch cases

rng193 committed:
linting

rng193 committed:
update linting on the nav component

raymondng1 commented:
1. Changed the linting to the 2 spaces instead of the tabs. 
2. The Navigation component into one and switched from the ternary operators, to the "switch cases" so a bit easier to read.
3. There is also an 'add product' route to the navigation bar if the user type is an Admin when they log in, as well as anybody's user profile with their name on it. 
4. Updated the login component as well with @ShelOrock toast! Thank you :)

ShelOrock reviewed (approved):
can't wait to see this working!
